code/modules/surgery/organs/organ.dm

```diff
@@ -245,6 +256,9 @@
 		add_attack_logs(user, owner, "Removed vital organ ([src])", !!user ? ATKLOG_FEW : ATKLOG_ALL)
 		owner.death()
 	owner = null
+	if(!QDELETED(src) && destroy_on_removal)
```

Comment:
If these are already qdeleted, we might just want to return null instead of src

Reply:
nit but I would would put the `destroy_on_removal` check on the left so we don't have to check `!QDELETED(src)` when a majority of organs are going to be `destroy_on_removal = FALSE`.